lib/charms/istio_k8s/v0/istio_info.py

```diff
@@ -0,0 +1,239 @@
+"""TODO: Add a proper docstring here.
+
+* using requirer:
+    * handle info changed event  <-- suggest a charm test to confirm this?
+
+* using provider:
+    * handle send error
+    * use the is_ready()
+
+
+requires cosl
+"""
+from typing import List, Optional
+
+from cosl.interfaces.utils import DatabagModelV2, DataValidationError
+from ops import CharmBase, Object, EventBase, CharmEvents, RelationDataAccessError, EventSource
+from pydantic import BaseModel, Field
+
+# The unique Charmhub library identifier, never change it
+LIBID = "17dff53d5bf649f29614365bde32451b"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+DEFAULT_RELATION_NAME = "istio-info"
+
+
+# Interface schema
+
+class IstioInfoAppData(DatabagModelV2, BaseModel):
+    """Data model for the istio-info interface."""
+
+    # TODO: Confirm that comparison between these objects works as expected so is_ready below works.
+
+    root_namespace: str = Field(
+        description="The root namespace for the Istio installation.",
+        examples=["istio-system"],
+    )
+
+    # To make it sortable, which is useful for comparing unordered sets like how relation data is stored.
+    # TODO: Move this to DatabagModelV2?
```

Comment:
I don't know why we can't put this into DatabagModelV2 instead, but just in case I mention it here too...

Reply:
removed this part of the code